pkg/validators/am0007_csv_install_modes.go

```diff
@@ -0,0 +1,72 @@
+package validators
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/utils"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+)
+
+func init() {
+	Registry.Add(AM0007)
+}
+
+var AM0007 = utils.Validator{
+	Code:        "AM0007",
+	Name:        "csv_install_modes",
+	Description: "Validate installMode is supported.",
+	Runner:      validateCSVInstallModes,
+}
+
+func validateCSVInstallModes(mb utils.MetaBundle) (bool, string, error) {
+	installMode := mb.AddonMeta.InstallMode
+
+	for _, bundle := range mb.Bundles {
+		bundleName, err := utils.GetBundleNameVersion(bundle)
+		if err != nil {
+			return false, "", err
+		}
+		spec, err := extractCSVSpec(bundle)
+		if err != nil {
+			return false, "", fmt.Errorf("Can't extract CSV for %v, got %v.", bundleName, err)
+		}
+		if success, failureMsg := isInstallModeSupported(spec.InstallModes, installMode); !success {
+			return false, fmt.Sprintf("Bundle %v failed CSV validation: %v.", bundleName, failureMsg), nil
+		}
+	}
+	return true, "", nil
+}
+
+func isInstallModeSupported(installModes []operatorsv1alpha1.InstallMode, target string) (bool, string) {
+	var allSupported []string
+	for _, im := range installModes {
+		imType := string(im.Type)
+		if im.Supported {
+			allSupported = append(allSupported, imType)
+		}
+
+		if imType == target {
+			return im.Supported, ""
+		}
+	}
+	return false, fmt.Sprintf("Target installMode %v is not supported. CSV only supports these installModes %v.", target, allSupported)
```

Comment:
So as to generate a relevant failureStr instead of straight away returning at `return im.Supported, ""` (Original Line: 52)

```suggestion
	targetImSupported := false
	for _, im := range installModes {
		imType := string(im.Type)
		if im.Supported {
			allSupported = append(allSupported, imType)
		}

		if imType == target {
			targetImSupported = im.Supported 
		}
	}
	failureStr := ""
	if !targetImSupported {
	    failureStr = fmt.Sprintf("Target installMode %v is not supported. CSV only supports these installModes %v.", target, allSupported)
	}
	return targetImSupported, failureStr 
```

Reply:
Good catch, if `im.Supported` is false we would get an empty failure message. :)